xhr/client/src/main/scala/endpoints4s/xhr/Endpoints.scala

```diff
@@ -294,45 +294,49 @@ trait EndpointsWithCustomErrors
       request: Request[A],
       response: Response[B],
       a: A
-  )(
-      onload: Either[Throwable, B] => Unit,
-      onerror: XMLHttpRequest => Unit
-  ): Unit = {
+  ): (js.Promise[B], Unit => Unit) = {
```

Comment:
Here too, the fact that we always create a `Thenable` which we then convert to something else seems a bit weird.  Would it be possible to make the transformation of the “callback-based” version to the “future-like” version in every concrete interpreter? Maybe it does not really matter, actually, but I was just wondering…

Reply:
I can see the yuckiness of mandatory intermediary `Promise`, I need to think about it for a bit.

Reply:
Did I get your idea right?